extras/lbf-nix/lbf-rust.nix

```diff
@@ -0,0 +1,154 @@
+pkgs:
+# LambdaBuffers Frontend
+lbf:
+# LambdaBuffers Rust Codegen
+lbg-rust:
+let
+  lbfRustOpts =
+    {
+      # Source that is passed to `lbf` as the `--import-path` flag and used to find `files`.
+      # Examples: src = ./api
+      src
+    , # Additional sources that are passed to `lbf` as the `--import-path` flag.
+      # Examples: imports = { lbf-prelude = ./lbf-prelude; }
+      imports ? { }
+    , # .lbf files in `src` to compile and codegen.
+      # Examples: files = [ "Foo.lbf" "Foo/Bar.lbf" ]
+      files
+      # Classes for which to generate implementations for (default lbf-prelude classes).
+    , classes ? [ ]
+    , # Dependencies to include in the Cabal's `build-depends` stanza.
+      # examples: dependencies = [ "lbf-prelude" ]
+      dependencies ? [ ]
+    , configs ? [ ]
+    , # Name of the package and also the name of the Cabal package.
+      # Examples: name = "lbf-myproject"
+      name
+    , # Version of the package and also the version of the Cabal package.
+      # Examples: version = "0.1.0.0"
+      version ? "0.1.0"
+    }: { inherit src imports files classes dependencies configs name version; };
+
+  lbf-build = import ./lbf-build.nix pkgs lbf;
+
+  lbfBuild = opts: with (lbfRustOpts opts);
+    let
+      findModules = root: map
+        (path: builtins.replaceStrings [ "/" ] [ "." ]
+          (pkgs.lib.strings.removePrefix "./" (pkgs.lib.strings.removeSuffix ".lbf"
+            (pkgs.lib.path.removePrefix root path))))
+        (builtins.filter (pkgs.lib.hasSuffix ".lbf")
+          (pkgs.lib.filesystem.listFilesRecursive root));
+      packageSet =
+        pkgs.writeTextFile {
+          name = "lb-packages";
+          text =
+            builtins.toJSON
+              ({ crate = findModules src; } // builtins.mapAttrs (_: findModules) imports);
+        };
+
+    in
+    lbf-build.build
+      {
+        inherit src;
+        opts = {
+          inherit files;
+          import-paths = pkgs.lib.attrsets.attrValues imports;
+          gen = lbg-rust;
+          gen-classes = classes;
+          gen-dir = "autogen";
+          gen-opts = [ "--packages=${packageSet}" ] ++ builtins.map (c: "--config=${c}") configs; # WARN(bladyjoker): If I put quotes here everything breaks.
+          work-dir = ".work";
+        };
+      };
+
+  cargoTemplate = opts: with (lbfRustOpts opts);
+    pkgs.writeTextFile {
+      name = "lambda-buffers-cabal-template";
+      text = ''
+        [package]
+        name = "${name}"
+        version = "${version}"
+        edition = "2021"
+
+        [dependencies]
+      '';
+    };
+
+  # 
+  crateVersions = pkgs.writeTextFile {
+    name = "lambda-buffers-crate-versions";
+    text = ''
+      num-bigint = "0.4.4"
+      serde_json = { version = "1.0.107", features = ["arbitrary_precision"] }
+      plutus-ledger-api = { git = "https://github.com/mlabs-haskell/plutus-ledger-api-rust", features = [ "lbf", ], rev = "fb93fa590908580eb40368369bf6614d42ce9a95" }
+    '';
```

Comment:
This is a temporary place for this, should we put this into the `bulid.nix` file?

Reply:
I don't understand, can you elaborate?

Reply:
Why is this hardcoded? Isn't this something codegen gives you? and you just fill it in?

Reply:
with codegen, we only get the dependency names (e.g. `num-bigint`), but not the versions
what I'm doing a few lines below is kind of a mapping with grep, so in the end, if a package does not require `num-bigint`, it won't be included.
I'm not sure where this information should be: we could inject it into the codegen, or it could be a solely nix thing

Reply:
I'm creating an issue for this, I need to think about where to put these default versions.

Reply:
https://github.com/mlabs-haskell/lambda-buffers/issues/167